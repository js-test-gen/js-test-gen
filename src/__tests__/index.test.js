import { generateTest, generateTestTemplate } from "../index";

const fileDetails = {
  srcFileName: "index",
  importFromPath: ".."
};
const namedSnippet = "export const add1 = (x) => x + 1";
const defaultSnippet = `
const obj = {};
export default obj`;
const testFileSnippet = `
${namedSnippet} 
${defaultSnippet}`;

describe("generateTest", () => {
  it("should return an empty string if no params passed", () => {
    expect(generateTest()).toBe("");
  });
  it("should return an empty string if passed no js contents", () => {
    expect(generateTest("Invalid js contents")).toBe("");
  });
  it("should return an empty string if passed no js contents ", () => {
    expect(generateTest(namedSnippet)).toMatchSnapshot();
  });
});
describe("generateTestTemplate", () => {
  it("should return an empty string if no params passed", () => {
    expect(generateTestTemplate()).toBe("");
  });
  it("should return an empty string if passed no js contents", () => {
    expect(generateTestTemplate("Invalid js contents")).toBe("");
  });
  it("should return a test file template if js contents contains default & named mods", () => {
    expect(
      generateTestTemplate({
        contents: testFileSnippet,
        ...fileDetails
      })
    ).toMatchSnapshot();
  });
  it("should return a named mod template if template contains no default mod", () => {
    expect(
      generateTestTemplate({
        contents: namedSnippet,
        ...fileDetails
      })
    ).toMatchSnapshot();
  });
  it("should return a default mod template if template contains no named mods", () => {
    expect(
      generateTestTemplate({
        contents: defaultSnippet,
        ...fileDetails
      })
    ).toMatchSnapshot();
  });
});
