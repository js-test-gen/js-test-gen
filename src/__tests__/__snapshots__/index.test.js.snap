// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateTest should return an empty string if passed no js contents  1`] = `
"describe(\\"add1\\", () => {
  it(\\"should fail auto generated test\\", () => {
    expect(add1()).toBe(false);
  });
});
"
`;

exports[`generateTestTemplate should return a default mod template if template contains no named mods 1`] = `
"import obj from \\"..\\";
describe(\\"obj\\", () => {
  it(\\"should fail auto generated test\\", () => {
    expect(obj()).toBe(false);
  });
});
"
`;

exports[`generateTestTemplate should return a named mod template if template contains no default mod 1`] = `
"import { add1 } from \\"..\\";
describe(\\"add1\\", () => {
  it(\\"should fail auto generated test\\", () => {
    expect(add1()).toBe(false);
  });
});
"
`;

exports[`generateTestTemplate should return a test file template if js contents contains default & named mods 1`] = `
"import obj, { add1 } from \\"..\\";
describe(\\"obj\\", () => {
  it(\\"should fail auto generated test\\", () => {
    expect(obj()).toBe(false);
  });
});
describe(\\"add1\\", () => {
  it(\\"should fail auto generated test\\", () => {
    expect(add1()).toBe(false);
  });
});
"
`;
